uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Upgrader/Step/1030.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+use CRM_HRLeaveAndAbsences_BAO_LeaveRequest as LeaveRequest;
+
+trait CRM_HRLeaveAndAbsences_Upgrader_Step_1030 {
+  
+  /**
+   * Disables or deletes existing "sickness reason" option values
+   * and added new ones
+   *
+   * @return bool
+   */
+  public function upgrade_1030() {
+    $oldSicknessReasons = civicrm_api3('OptionValue', 'get', [
+      'return' => ['name'],
+      'option_group_id' => 'hrleaveandabsences_sickness_reason',
+    ]);
+  
+    $result = $this->up1030_getLeaveRequestTotal();
+    if ($result == 0) {
+      $this->up1030_deleteOldSicknessReasons($oldSicknessReasons['values']);
+    }
+    else {
+      $this->up1030_disableOldSicknessReasons($oldSicknessReasons['values']);
+    }
+    
+    $this->up1030_addNewSicknessReasons();
+    
+    return TRUE;
+  }
+  
+  /**
+   * Counts the total number of leave requests related to sickness reasons
+   *
+   * @return int
+   */
+  private function up1030_getLeaveRequestTotal() {
+    $leaveRequest = new LeaveRequest();
+    $leaveRequest->whereAdd('sickness_reason IS NOT NULL');
+    $leaveRequest->find();
+  
+    // N is Number of rows returned from a query
+    return $leaveRequest->N;
+  }
+  
+  /**
+   * Disables existing "sickness reason" option values
+   *
+   * @param array $sicknessReasons
+   */
+  private function up1030_disableOldSicknessReasons($sicknessReasons) {
+    foreach ($sicknessReasons as $id => $sicknessReason) {
+      civicrm_api3('OptionValue', 'create', [
+        'id' => $id,
+        'is_active' => 0
+      ]);
+    }
+  }
+  
+  /**
+   * Deletes existing "sickness reason" option values
+   *
+   * @param array $sicknessReasons
+   */
+  private function up1030_deleteOldSicknessReasons($sicknessReasons) {
```

Comment:
Same as above, I'm not sure "old" means anything. In fact both this method and `up1030_disableOldSicknessReasons` would probably be clearer just as `up1030_disableOptionValues` and `up1030_deleteOptionValues`

Reply:
Noted. I will update appropriately.